258 Splash Screen

najimgh opened the pull request:
Added a Splash Screen which will be shown before the Assets Page.

This resolves #258

najimgh committed:
Issue (#258)
- Added Splash Screen FXML
- Added Splash Screen Controller
- Modified CBMSApplication.java to start on the Splash Screen

PaulMicu commented on JavaFXUserInterface/src/main/java/Controllers/SplashScreenController.java:
```diff
@@ -0,0 +1,69 @@
+        @Override
+        public void run() {
+            try {
+                Thread.sleep(5500); //Displays the SplashScreen for 5.5 seconds
```
I dont think there should be a hardcoded amount of time the splash screen is displayed, it should be displayed for as long as the app takes to boot up, no longer, no shorter

najimgh replied:
Splash Screen is now implemented using a preloader with a fade transition. The Splash Screen is displayed as long as all of the assets are not loaded completely.

jeffbuonamici commented on JavaFXUserInterface/src/main/resources/SplashScreen.fxml:
```diff
@@ -0,0 +1,33 @@
+      </HBox>
+      <HBox alignment="CENTER">
+         <children>
+            <Label text="M i n e r v a" textAlignment="CENTER" textFill="WHITE">
+               <font>
+                  <Font name="Futura Medium" size="55.0"/>
+               </font>
+            </Label>
```
Text Alignment, Text fill, font family, and font size should just be in the css

najimgh replied:
Fixed

jeffbuonamici commented on JavaFXUserInterface/src/main/resources/css/style.css:
```diff
@@ -67,4 +67,9 @@
+}
+
+.splash-screen-bg {
+    -fx-background-color: rgb(22, 53, 96);
+    -fx-background-color: linear-gradient(to top right, rgba(22, 53, 96, 1) 0%, rgba(63, 143, 204, 1) 85%);;
```
Is there a reason why there's two background-color attributes?

najimgh replied:
The extra background-color attribute has been removed.

najimgh committed:
Issue (#258)
- Moved styles on the splash screen title to the style.css file
- Removed extra background-color property

najimgh committed:
Issue (#258)
- The splash screen is now using a Preloader instead of a fixed time to be displayed.
- Splash Screen image changed.

najimgh committed:
Issue (#258)
- Added try/Catch when loading the assets

najimgh committed:
Issue (#258)
- Removed observableList when loading assets

najimgh committed:
Issue (#258)
- Added Comments
- Changed the Directory of the SplashScreenPreloader

najimgh committed:
Delete SplashScreenController.java

jeffbuonamici reviewed (approved):
My concerns were addressed. Approved.
